### ISSUE TITLE
"pluginContext.vueCompilerOptions.plugins is not iterable" error even with propsDestructure = true

### ISSUE THREAD

**Issue by adamdehaven** [nuxt/nuxt#29144]
### Environment

- Operating System: Darwin
- Node Version: v20.17.0
- Nuxt Version: 3.13.1
- CLI Version: 3.13.2
- Nitro Version: 2.9.7
- Package Manager: pnpm@9.10.0

### Reproduction

Unsure what to provide 🫣 

### Describe the bug

Referencing [this comment](https://github.com/nuxt/nuxt/issues/28829#issuecomment-2329411615) from https://github.com/nuxt/nuxt/issues/28829, adding `propsDestructure: true` to the `nuxt.config.ts` should resolve `pluginContext.vueCompilerOptions.plugins is not iterable`, and should be resolved as of the latest release via [this comment](https://github.com/nuxt/nuxt/issues/28856#issuecomment-2333761711); however, the issue still persists.

**Comment by danielroe** [comment]
I don't believe this is related to `propsDestructure`.

**Comment by adamdehaven** [comment]
🤔 the error goes away if I remove propsDestructure

**Comment by adamdehaven** [comment]
When I remove the `propsDestructure` I see the following errors when running the nuxt dev server: 

```
Cannot find type definition file for '.vue-global-types/vue_3.5_false.d.ts'.
```

![image](https://github.com/user-attachments/assets/d490a4fe-da4b-4dbc-8255-df8ddc7127d2)

The warnings appear to be originating from `nuxt-og-image` and some of the `@nuxtjs/seo` packages.

Might be sending you on a wild 🪿 chase

**Comment by adamdehaven** [comment]
Here's the error for the actual "not iterable" output

![image](https://github.com/user-attachments/assets/ff3e3789-d01d-4da7-8413-d6a2b545e667)

**Comment by adamdehaven** [comment]
I'm still not sure why I'm getting the `not iterable` error though; it's unrelated and definitely is linked to the latest release somehow 🤔

**Comment by danielroe** [comment]
It's more likely to be linked to your version of vue-tsc. Are you using build time type checking?

**Comment by adamdehaven** [comment]
> It's more likely to be linked to your version of vue-tsc or vite-plugin-checker. Are you using build time type checking?

Yes, during dev and build. 

```
vue-tsc: 2.1.6
vite-plugin-checker: 0.7.2
```

**Comment by adamdehaven** [comment]
(In my monorepo) if I add a pnpm override and force `vite-plugin-checker` to `0.8.0` the issue is resolved. Looks to be referenced by `@nuxt/vite-builder: 3.13.1`

```json
"pnpm": {
    "overrides": {
      "vite-plugin-checker": "0.8.0"
    }
  },
```

**Comment by danielroe** [comment]
Should be resolved in v3.13.2 then.

**Comment by adamdehaven** [comment]
> Should be resolved in v3.13.2 then.

Well see the problem there for me is 😅 https://github.com/nuxt/nuxt/issues/29116

**Comment by danielroe** [comment]
Well, let's solve that one too!

Marking this one as resolved for now.

**Comment by adamdehaven** [comment]
> When I remove the `propsDestructure` I see the following errors when running the nuxt dev server:
> 
> ```
> Cannot find type definition file for '.vue-global-types/vue_3.5_false.d.ts'.
> ```
> 
> ![image](https://github.com/user-attachments/assets/d490a4fe-da4b-4dbc-8255-df8ddc7127d2)
> 
> The warnings appear to be originating from `nuxt-og-image` and some of the `@nuxtjs/seo` packages.
> 
> Might be sending you on a wild 🪿 chase

For anyone coming along later, this particular issue is resolved in `vue-tsc@2.1.8` via  https://github.com/vuejs/language-tools/pull/4924
